provider/cmd/pulumi-resource-xyz/main.go

```diff
@@ -12,20 +12,66 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-//go:generate go run ./generate.go
 package main
 
 import (
-	_ "embed"
-	"github.com/pulumi/pulumi-xyz/provider/pkg/provider"
-	"github.com/pulumi/pulumi-xyz/provider/pkg/version"
-)
+	"math/rand"
+	"time"
 
-var providerName = "xyz"
+	p "github.com/pulumi/pulumi-go-provider"
+	"github.com/pulumi/pulumi-go-provider/infer"
+)
 
-//go:embed schema-embed.json
-var pulumiSchema []byte
+// Version is initialized by the Go linker to contain the semver of this build.
+var Version string = "0.0.1"
```

Comment:
This looks cleaner, and the value should never be read anyway.
```suggestion
var Version string
```